### PR TITLE
Update for futures-preview alpha 11 and nightly-2018-12-26

### PR DESCRIPTION
**Pull request by zippy**
currently this comments out the async timeouts in:
`core/src/network/actions/get_entry.rs` and `core/src/network/actions/custom_sent.rs` because I'm not sure how to get them to work given the new changes, but I don't think we have any tests that test them either.

Adding that to the SoA as a follow up.

### CODE REVIEW

File: core/src/dht/actions/hold.rs [hunk 1]
```diff
@@ -37,8 +34,6 @@ pub struct HoldEntryFuture {
     address: Address,
 }
 
-impl Unpin for HoldEntryFuture {}
```
Comment (maackle) [inline core/src/dht/actions/hold.rs]: @zippy why is this removed? Assuming the compiler thinks it's no longer necessary?

File: core/src/dht/actions/remove_entry.rs [hunk 2]
```diff
@@ -40,8 +40,6 @@ pub struct RemoveEntryFuture {
     action: ActionWrapper,
 }
 
-impl Unpin for RemoveEntryFuture {}
```
Comment (maackle) [inline core/src/dht/actions/remove_entry.rs]: Same, why removed?

File: core/src/dht/actions/hold.rs (hunk 1 above)
Comment (zippy) [reply core/src/dht/actions/hold.rs]: I had to.  Apparently Unpin became a private element (the use statement broke and I had to remove it), and gets auto-implemented in the default case.  I didn't dig too deep, just got it to work.
